tests/conftest.py

```diff
@@ -15,3 +15,108 @@ def gauss_spectrum(wavelength):
     width = (wavelength.max() - wavelength.min()) / 6
     sp = np.exp(-(wavelength - centre) ** 2 / width)
     return sp, interp1d(wavelength, sp, bounds_error=False, fill_value=0)
+
+
+@fixture
+def np_junction():
```

Comment:
This two fixtures are nearly identical and the only practical difference is the order of the dopants. I would put all of this (with some minor changes) in a normal function (not a `fixture`) just above this one, that takes as inputs the Na and Nd doping for the top and bottom layers:

```python
def junction(nd_top=0, na_top=0, nd_bottom=0, na_bottom=0)
```
Then the two fixtures just call this function with:

```python
@fixture
def np_junction()
    Nd=...
    Na=...
    return junction(nd_top=Nd, na_bottom=Na)

@fixture
def pn_junction()
    Nd=...
    Na=...
    return junction(nd_bottom=Nd, na_top=Na)
```